include/splitvector/splitvec.h

```diff
@@ -154,49 +167,29 @@ class SplitVector {
     */
    HOSTONLY T* _allocate_and_construct(size_t n, const T& val) {
       T* _ptr = _allocator.allocate(n);
-      for (size_t i = 0; i < n; i++) {
-         _allocator.construct(&_ptr[i], val);
+      if constexpr (!std::is_trivially_copy_constructible_v<T> ){
+         for (size_t i = 0; i < n; i++) {
+            _allocator.construct(&_ptr[i], val);
```

Comment:
There's now a mix of using both calls to `_allocator` and calls to `split_gpuMallocManaged` etc. Worth consolidating I think?